openstates/de/legislators.py

```diff
@@ -68,64 +92,64 @@ def scrape_bio(self, term, chamber, district, name, url):
-        email = doc.xpath(xpath)[0].text_content()
-
-        # If multiple email addresses listed, only take the official
-        # noone@state.de.us address.
-        emails = re.split(r'(?:\n| or |;|\s+)', email)
-        for email in filter(None, emails):
-            if email.strip():
-                info['email'] = email.strip()
-                break
+        email = doc.xpath(".//a[contains(@href,'mailto')]")
+        email = email[0].text_content().strip()
+        if not email:
+            email = None
 
         # Offices
-        offices = []
-        for (td, name, type_) in data:
-            office = dict(name=name, type=type_, phone=None,
-                          fax=None, email=None, address=None)
-
-            chunks = td.text_content().strip().split('\n\n')
-            chunks = [s.strip() for s in chunks]
-            chunks = filter(None, chunks)
-            if len(chunks) == 1:
-                if ':' in chunks[0].splitlines()[0]:
-                    # It's just phone numbers with no actual address.
-                    numbers = [chunks[0]]
-                    office['address'] = None
-                elif chunks[0].strip().startswith("("):
-                    numbers = [chunks[0]]
-                    office['address'] = None
+
+        leg_office = dict(name="Capitol Office", type="capitol",
+                        phone=None, fax=None, email=email, address=None)
+        dist_office = dict(name="District Office", type="district",
+                        phone=None,fax=None, email=email, address=None) 
+
+        #this is enormously painful, DE.
+        office_list = doc.xpath("//tr")
+        for office in office_list:
+            title_td = 0
+            #in some trs the photo is the first td
+            if len(office.xpath("./td/img")) > 0:
+                title_td = 1
+            try:
+                title_text = office.xpath("./td")[title_td].text_content().lower()
+                content = office.xpath("./td")[title_td+1].text_content()
+
+            except IndexError:
+                continue
+
+            if "legislative office" in title_text:
+                leg_office["address"] = content.strip()
+            if "legislative phone" in title_text:
+                phones = content.lower().split("\n")
+                if len(phones) == 1:
+                    phone = phones[0].replace("phone:","").strip()
+                    if phone:
+                        leg_office["phone"] = phone
+                else:
+                    for line in phones:
+                        if "phone" in line:
+                            leg_office["phone"] = line.replace("phone:","").strip()
+                        elif "fax" in line:
+                            leg_office["fax"] = line.replace("fax:","").strip()
+
+            if "outside office" in title_text:
+                dist_office["address"] = content.strip()
+            if "outside phone" in title_text:
```

Comment:
AFAICT, this is identical to the `"legislative phone"` code; could it be put into a dedicated function?